### ISSUE TITLE
ebs_optimized in resource_aws_launch_template doesn't accept string true/false

### ISSUE THREAD

**Issue by kinghuang** [terraform-providers/terraform-provider-aws#5994]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

Terraform v0.11.8
+ provider.aws v1.37.0

### Affected Resource(s)

* aws_launch_template

### Terraform Configuration Files

```hcl
resource "aws_launch_template" "managers" {
  name = "${var.name}-managers"
  image_id      = "${data.aws_ami_ids.moby.ids[0]}"
  instance_type = "${var.manager_instance_type}"

  …

  block_device_mappings {
    device_name = "/dev/xvdb"
    ebs {
      volume_type = "${var.manager_disk_type}"
      volume_size = "${var.manager_disk_size}"
    }
  }
  ebs_optimized = "true"
}
```

### Debug Output

```
Error: module.swarm.aws_launch_template.managers: expected ebs_optimized to be
one of ["", false, true], got true
```

### Panic Output

N/A

### Expected Behavior

`ebs_optimized` should be true.

### Actual Behavior

An error occurred using the string `"true"` for `ebs_optimized`.

### Steps to Reproduce

1. `terraform apply`

### Important Factoids

While upgrading from version 1.35.0 to 1.37.0 of the AWS provider, I ran into this error with the `ebs_optimized` parameter in the `aws_launch_template` resource. I'm using the string value `"true"`, but that resulted in the error that the value was expected to be one of `["", false, true]`. Changing the value to boolean value `true` worked.

From 1.35.0 to 1.36.0, the value type for ebs_optimized change from `schema.TypeBool` to `schema.TypeString` (https://github.com/terraform-providers/terraform-provider-aws/commit/eeb04c6a07450348b92bfa80e904f73778b5ccf8), and uses `validateTypeStringNullableBoolean` for ValidateFunc. The new validation function doesn't seem to allow for string `"true"` and `"false"` values.

### References

Change of `ebs_optimized`'s value type and validation function: https://github.com/terraform-providers/terraform-provider-aws/commit/2e8a2b3b17a56b028af497aafd3a36ed15e513e1
Addition of validateTypeStringNullableBoolean: #5632

**Comment by bflad** [comment]
Hi @kinghuang 👋  Sorry for the trouble. We generally try to prevent configuration regressions such as this although the usage of the string versions of `"true"`/`"false"` is less likely to be accounted for.

I think we can safely add `"false"` and `"true"` to the new validation function, I'm working on verifying that.

**Comment by bflad** [comment]
The fix for this has been merged and will release with version 1.38.0 of the AWS provider, likely later today.

**Comment by kinghuang** [comment]
Awesome. Thanks for the quick turnaround!

**Comment by bflad** [comment]
This has been released in [version 1.38.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1380-september-26-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
